### PR TITLE
Rework the way companies DB is composed

### PR DESCRIPTION
**Pull request by ameshkov**
This pull request changes the way companies DB is composed.

Now there's a clear distinction between the source data and the compiled data.
We can change what's in `source`, `dist` will be built automatically.

Second, it now merges whotracksme and AG data automatically (and validates it at the same time).

We should change AdGuard DNS and AdGuard Home to use `dist/trackers.json` and `dist/companies.json` as the only sources of information. Note, that if the data comes from AdGuard, it is labeled with the `source` property (it's valid for both `trackers.json` and `companies.json`). We use this information in AdGuard Home.

### CODE REVIEW

**Review by ainar-g — APPROVED** [review]
@ameshkov, the only change relevant to AdGuard Home I see that `dist/adguard.json` has become `dist/trackers.json`?

**Comment by ameshkov** [comment]
@ainar-g nope, not really.

You now do not need `dist/whotracksme.json` as all information is now merged into `dist/trackers.json`.

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (slavaleleka) [inline .vscode/launch.json:1]: why `.vscode` is not added to `.gitignore`?

Comment (ainar-g) [reply .vscode/launch.json:1]: @slavaleleka, or, better, to the developer's personal global gitignore :grin:.

<https://stackoverflow.com/a/7335487/1892060>

**Comment by ainar-g** [comment]
@ameshkov, is it okay that most objects don't have a `source` field then?  Feels a bit weird, but if that's the format then okay.

File: index.js [hunk 2]
```diff
@@ -1,131 +1,255 @@
...
+    const companiesData = {
+        timeUpdated: new Date().toISOString(),
+        companies: {},
     };
 
-    const whotracksme = {
+    consola.info(`Reading ${COMPANIES_INPUT_PATH}`);
+    const companies = JSON.parse(fs.readFileSync(COMPANIES_INPUT_PATH).toString());
+
+    // Copy whotrackme companies and merge with adGuard companies
```
Comment (slavaleleka) [inline index.js:148]: `adGuard`

**Comment by ameshkov** [comment]
> @ameshkov, is it okay that most objects don't have a source field then? Feels a bit weird, but if that's the format then okay.

@ainar-g yeah, this way the size will be smaller.

**Review by atropnikov — APPROVED** [review]

**Pull request merged**